---
title: "`r read.dcf('DESCRIPTION', fields = 'Package')[1]`"  
author: "`r rworkflows::use_badges()`" 
date: "<h4>README updated: <i>`r format( Sys.Date(), '%b-%d-%Y')`</i></h4>"
output:
  github_document
---

<!-- To modify Package/Title/Description/Authors fields, edit the DESCRIPTION file -->

```{r, echo=FALSE, include=FALSE}
pkg <- read.dcf("DESCRIPTION", fields = "Package")[1]
title <- read.dcf("DESCRIPTION", fields = "Title")[1]
description <- gsub("\n"," ",read.dcf("DESCRIPTION", fields = "Description")[1])
URL <- read.dcf('DESCRIPTION', fields = 'URL')[1]
owner <- strsplit(URL,"/")[[1]][4]
repo <-strsplit(URL,"/")[[1]][5]
```
 
## Introduction

Many genes have been associated with diseases 
Multi-Scale Target Explorer (`MSTExplorer`) systematically identifies, prioritises, and visualises cell-type-specific gene therapy targets across the phenome. 

Core functionalities include:  

**1. Conducting phenotype x cell type genetic association tests at scale**
  
  - The [Human Phenotype Ontology](https://hpo.jax.org/) (integrated with gene annotations from [OMIM](https://omim.org/) / [DECIPHER](https://www.deciphergenomics.org/) / [ORPHANET](https://www.orpha.net/)) is used as the source of phenotype gene signatures. Each gene-phenotype associated is given a continuous score that approximates the current strength of evidence for the association (using data derived from [GenCC](https://thegencc.org/)).  
    
  - Whole-body scRNA-seq atlases from humans (across multiple developmental stages) are used as a data-driven source of cell type-specific gene markers.  
  - The underlying association tests are designed for both speed and accuracy using memory-efficient data structures, and a highly parallelisable implementation of Generalised Linear Regression (GLM). For example, associations for all pairwise combinations of >11k phenotypes x >200 cell types (>2,200,000 associations) can be in <30 minutes on a Macbook laptop with 10 CPU cores).

**2. Inferring multi-scale causal graphs of disease**

`MSTExplorer` allows users to easily infer and construct multi-scale causal graphs of Diseases (blue nodes) -> Phenotypes (purple nodes) -> Cell types (orange nodes) -> Genes (yellow nodes).  

![*Example multi-scale network focused on lethal skeletal dysplasia, a phenotype of multiple diseases*](https://github.com/neurogenomics/rare_disease_celltyping/blob/299abe0ccd00644bc2f05a1389704fe196a3e868/manuscript/_manuscript/img/fig-therapy-examples-supp/lethal_skeletal_dysplasia.png?raw=true){height=400px}

[See here for more example networks.](https://github.com/neurogenomics/rare_disease_celltyping/tree/299abe0ccd00644bc2f05a1389704fe196a3e868/manuscript/_manuscript/img/fig-therapy-examples-supp).

**3. Prioritising cell-type-specific gene therapy targets**

`MSTExplorer` also provides a comprehensive and customisable pipeline that can be run via a single function (`prioritise_targets()`) to produce the most promising cell-type-specific gene therapy targets across the phenome.

## Installation 

Within R: 

```R
if(!require("BiocManager")) install.packages("BiocManager")

BiocManager::install("`r paste(owner,repo,sep='/')`")
library(`r pkg`)
```

## Documentation 

#### [Website](https://`r owner`.github.io/`r pkg`) 
#### [Get started](https://`r owner`.github.io/`r pkg`/articles/`r pkg`)
#### [Docker/Singularity Container](https://`r owner`.github.io/`r pkg`/articles/docker.html)

## Citation

If you use ``r pkg``, please cite: 

<!-- Modify this by editing the file: inst/CITATION  -->
> `r utils::citation(pkg)$textVersion`

## Contact

### [Neurogenomics Lab](https://www.neurogenomics.co.uk)   
UK Dementia Research Institute  
Department of Brain Sciences  
Faculty of Medicine  
Imperial College London  
[GitHub](https://github.com/neurogenomics)

## Session Info 

<details> 

```{r Session Info}
utils::sessionInfo()
```

</details>  
<hr>
